### ISSUE TITLE
Component: Masthead: Accessibility - issues related to hamburger menu once it is expanded

### ISSUE THREAD

**Issue by Praveen-sr** [carbon-design-system/ibm-dotcom-library#2058]
## Detailed description

Issues :- 
1. Close icon - screen reader is reading the label as open menu button
2. After expanding the Hamburger menu visually there is close button, IBM logo link and the options but screen reader is reading content which is not visually available.
For example user profile and search edit combo box. 

Issues on Web components :
1. Hamburger menu - There is no label for the hamburger menu, , screen reader is announcing as button without any label.
2. IBM logo link - screen reader is announcing the mentioned control as  " link " and is not operable.

Expected :-
1. Give appropriate label.
2. screen reader should not read visually hidden components

Expected results for web components :
1. Provide a descriptive label , a generic example would be  "side Navigation"
2. screen reader should announce appropriate label/ link text. Make the logo operable.

**Comment by RobertaJHahn** [comment]
@kennylam I am closing this bug based on the MegaMenu discussions in the 7/13 QA meeting.
